test/test_torch.py

```diff
@@ -645,6 +645,11 @@ def test_cross_validation(self):
                 IndexError, "Dimension out of range",
                 lambda: torch.cross(torch.rand(5, 3, 4), torch.rand(5, 3, 4), dim=-5))
 
+        def test_ones_overflow(self):
+            with self.assertRaisesRegex(
+                    RuntimeError, r'int64 overflow while computing numel for sizes \[2, 2, 1, 2305843009213693952\]'):
+                torch.ones(2, 2, 1, 2305843009213693952)
```

Comment:
Would you add one of the cases described in the issue, too, to be sure it, in particular, doesn't regress? Like `torch.nn.functional.adaptive_avg_pool1d(torch.randn([2,2,2]), 9132760301568586890)`.